packages/rich-text/src/plugins/Tracking/createTrackingPlugin.ts

```diff
@@ -111,5 +111,43 @@ export const createTrackingPlugin = (onAction: RichTextTrackingActionHandler): R
 
       return editor;
     },
+    then: (editor) => ({
+      editor: {
+        insertData: {
+          format: 'text/html',
+          getFragment: ({ data }) => {
+            let source;
+
+            if (data.includes('docs-internal-guid')) {
+              source = 'Google Docs';
+            }
+
+            if (data.includes('<google-sheets-html-origin>') || data.includes('data-sheets-')) {
+              source = 'Google Spreadsheets';
+            }
+
+            if (data.includes('class="TextRun') || data.includes('class="OutlineElement')) {
```

Comment:
This feels too generic. Is the only way we can tell it's MS Word Online? Also what about normal MS Word and Excel Sheets?

Reply:
I'd say none of that is reliable so maybe we can ignore the online version. The desktop MS Word however (I tested on Windows VM) adds a ` <meta name=ProgId content=Word.Document>` and ` <meta name=Generator content="Microsoft Word 15">`. Additionally, we also support the `TEXT/RTF` format specifically for MS Word

Reply:
Similar to Word, MS Excel on desktop adds ` <meta name=ProgId content=Excel.Sheet> <meta name=Generator content="Microsoft Excel 15">`

Reply:
Thanks for that! I will test if we can extract soem stuff from `TEXT/RTF` as well. I will bring up this question to our PH today. I still think it's safe to use one of those values. The context here is very specific, it happens only when pasting and it needs to be `text/html`, and values like those is not likely to be used  everywhere else.

Reply:
MS Word online doesn't use the `text/rtf` format so it's not helpful for that and since the desktop apps add specific metadata  I think it's ok to rely only on `text/html`.